### PR TITLE
Added Typescript React Base Setup

### PR DESCRIPTION
**Pull request by malteA**
Added a base component written in typescript.

### CODE REVIEW

File: snippets/TypescriptReactSetup.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
+### TypeScript React Compoent
```
Comment (flxwu) [inline snippets/TypescriptReactSetup.md]: typo 😉

File: snippets/TypescriptReactSetup.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+const htmlNode: HtmlElement = document.getElementById("demoTypescript");
+if (htmlNode) {
+    ReactDOM.render(<TypescriptComponent greeting="Hello 30 Seconds of Code" />, htmlNode);
+}
+```
+
+<!-- tags: 0,1 -->
+
+<!-- expertise: (0,1,2) -->
+<!-- Expertise levels:
```
Comment (flxwu) [inline snippets/TypescriptReactSetup.md]: please remove this template comment

File: snippets/TypescriptReactSetup.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+Here you are looking for a html element with the id "demoTypescript" and rendering the React Component inplace of it.
+
+```tsx
+const htmlNode: HtmlElement = document.getElementById("demoTypescript");
+if (htmlNode) {
+    ReactDOM.render(<TypescriptComponent greeting="Hello 30 Seconds of Code" />, htmlNode);
+}
+```
+
+<!-- tags: 0,1 -->
```
Comment (flxwu) [inline snippets/TypescriptReactSetup.md]: these are weird tags - should be "setup" or so :)

File: snippets/TypescriptReactSetup.md [hunk 4]
```diff
@@ -0,0 +1,45 @@
+### TypeScript React Compoent
+
+To get started with React and Typescript you need to import the React module, for Typescript you have to import the whole React Library.
```
Comment (flxwu) [inline snippets/TypescriptReactSetup.md]: just write that you need to import the React module. Also write the whole description as one block and put the whole code block below the description. See the 30-seconds-of-code snippets for reference :)

**Comment by skatcat31** [comment]
You deleted to much... You still need an expertise and taglist. Otherwise it LGTM

**Comment by malteA** [comment]
Ahh, i added the tags and expertise again. Thanks for the patience

**Comment by flxwu** [comment]
thanks for the contribution! @malteA

**Comment by flxwu** [comment]
sry, clicked on the wrong button 😅

**Pull request merged**